dimscord/helpers.nim

```diff
@@ -413,3 +414,26 @@ proc add*(component: var MessageComponent, item: SelectMenuOption) =
         "Can only add menu options to a SelectMenu."
     )
     component.options &= item
+
+
+proc waitForObject*[T: DimscordObject](client: DiscordClient, event: DispatchEvent,
```

Comment:
Could we perhaps make it `{.discardable.}`  instead of making another proc that returns `Future[void]` ?

Reply:
Sadly no.
Async procs are not allowed to be discardable